crates/bindings/README.md

```diff
@@ -0,0 +1,517 @@
+        assert_eq!(actual.field, i);
+    }
+}
+```
+
+#### Indices
+
+SpacetimeDB supports both single- and multi-column indexes. Currently, only [B-Tree](https://en.wikipedia.org/wiki/B-tree) indexes are supported.
+
+Indexes are declared using the syntax:
+
+[`#[table(index(name = my_index, btree(columns = [a, b, c]))]`](macro@crate::table#index).
+
+Multiple indexes can be declared, separated by commas.
+
+Single-column indices can also be declared using the
+
+[`#[index(btree)]`](macro@crate::table#indexbtree)
+
+column attribute.
+
+Any index supports getting a [`BTreeIndex`] using [`ctx`](crate::ReducerContext)`.db.{table}().{index}()`. For example,
+    `ctx.db.people().name()`.
+    
+[`BTreeIndex`] provides:
+ - [`BTreeIndex::filter`]
+ - [`BTreeIndex::delete`]
+
+## Reducers
+
+Reducers are declared using the [`#[reducer]` macro](macro@crate::reducer).
+
+`#[reducer]` is always applied to top level Rust functions. Arguments of reducers must implement [`SpacetimeType`]. Reducers can either return nothing, or return a `Result<(), E>`, where `E` implements `Debug`.
+
+```rust
+use spacetimedb::{reducer, ReducerContext};
+
+#[derive(Debug)]
+enum GiveItemError {
+    NoSuchItem(u64),
+    NonTransferable(u64)
+}
+
+#[reducer]
+fn give_player_item(
+    ctx: &ReducerContext,
+    player_id: u64,
+    item_id: u64
+) -> Result<(), GiveItemError> {
+    /* ... */
+}
+```
+
+Every reducer runs inside a [database transaction](https://en.wikipedia.org/wiki/Database_transaction). <!-- TODO: specific transaction level guarantees. --> This means that reducers will not observe the effects of other reducers modifying the database while they run. Also, if a reducer fails, all of its changes to the database will automatically be rolled back. Reducers can fail by [panicking](::std::panic!) or by returning an `Err`.
+
+#### The `ReducerContext` Type
+
+Reducers have access to a special [`ReducerContext`] argument. This argument allows reading and writing the database attached to a module. It also provides some additional functionality, like generating random numbers and scheduling future operations.
+
+The most important field of [`ReducerContext`] is [`.db`](ReducerContext#structfield.db). This field provides a view of the module's database. The [`#[table]`](macro@crate::table) macro generates traits that add accessor methods to this field.
+
+<!-- TODO: this seems to work sometimes, but not always... Sometimes the links to downstream trait implementations aren't generated for some reason. Maybe it only works in the same cargo workspace?
+
+To see all of the available methods on `ctx.db`, run `cargo doc` in your module's directory, and navigate to the `spacetimedb::Local` struct in the generated documentation. This will be at the path:
+- `[your_project_directory]/target/doc/spacetimedb/struct.Local.html` (non-Windows)
+- `[your_project_directory]\target\doc\spacetimedb\struct.Local.html` (Windows)
```

Comment:
I don't think you actually need to distinguish Windows vs Unix paths in 2025 Anno Domini. Just use `/` as a path separator.

Reply:
old habits